includes/WebAuthn/class-webauthn-keystore.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ *	Simple CRUD for keys
+ */
+class WebAuthnKeyStore {
+
+	const PUBKEY_USERMETA_KEY = '_two_factor_webauthn_pubkey';
+
+
+	/**
+	 * Array containing derived class instances
+	 */
+	private static $instance = null;
+
+	/**
+	 * Getting a singleton.
+	 *
+	 * @return object single instance of Core
+	 */
+	public static function instance() {
+
+		$class = get_called_class();
+
+		if ( is_null( self::$instance ) ) {
+			$args = func_get_args();
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 *	Prevent Instantinating
+	 */
+	private function __clone() { }
+
+	/**
+	 *	Protected constructor
+	 */
+	protected function __construct() {
+	}
+
+	/**
+	 * Get all keys for user
+	 *
+	 * @param int $user_id
+	 * @return array
+	 */
+	public function get_keys( $user_id ) {
+		return get_user_meta( $user_id, self::PUBKEY_USERMETA_KEY );
+	}
+
+	/**
+	 * Find specific key for user by
+	 *
+	 * @param int $user_id
+	 * @param string $keyLike
+	 * @return array|bool
+	 */
+	public function find_key( $user_id, $keyLike ) {
+		global $wpdb;
+
+		$found = $wpdb->get_results( $wpdb->prepare(
+			"SELECT * FROM $wpdb->usermeta WHERE user_id=%d AND meta_key=%s AND meta_value LIKE %s",
+			$user_id,
+			self::PUBKEY_USERMETA_KEY,
+			'%' . $wpdb->esc_like( $keyLike ) . '%'
+		) );
+		foreach ( $found as $key ) {
+			return maybe_unserialize( $key->meta_value );
+		}
+		return false;
+	}
+
+	/**
+	 * Check whether a key exists
+	 *
+	 * @param string $keyLike
+	 * @return bool
+	 */
+	public function key_exists( $keyLike ) {
+
+		global $wpdb;
+
+		$num_keys = $wpdb->get_var( $wpdb->prepare(
+			"SELECT COUNT(*) FROM $wpdb->usermeta WHERE meta_key=%s AND meta_value LIKE %s",
+			self::PUBKEY_USERMETA_KEY,
+			'%' . $wpdb->esc_like( serialize( $keyLike ) ) . '%'
+		) );
```

Comment:
Would it make sense to only use core functions like `get_user_meta( $user_id, self::PUBKEY_USERMETA_KEY )` and then doing the lookup on PHP? This would avoid uncached queries for sites that make use of object caching plugins.